src/test/java/org/broadinstitute/hellbender/engine/GATKToolUnitTest.java

```diff
@@ -945,4 +939,59 @@ private void writeHeaderAndBadVariant(final VariantContextWriter writer) {
         writer.add(vc);
     }
 
+    final String baseVariants = packageRootTestDir + "engine/feature_data_source_test.vcf";
+
+    @CommandLineProgramProperties(programGroup = TestProgramGroup.class, oneLineSummary = "GATKTool Intervals Test Walker", summary = "This is a test walker for GATKTool getIntervals")
+    private static class TestIntervalWalker extends GATKTool {
+        @Argument(fullName = StandardArgumentDefinitions.VARIANT_LONG_NAME, shortName = StandardArgumentDefinitions.VARIANT_SHORT_NAME, doc = "A VCF file containing variants", common = false, optional = false)
+        public String drivingVariantFile;
+
+        public TestIntervalWalker() {
+
+        }
+
+        @Override
+        public void traverse() {
+
+        }
+    }
+
+    @DataProvider(name = "TestGetIntervalsProvider")
+    public Object[][] getTestGetIntervalsProvider() {
+        return new Object[][]{
+                {"1:21-21", 1, hg19MiniReference, false},
+                {null, 4, hg19MiniReference, false},
+                {"1:21-21", 1, null, true},  //this will fail with an error
+                {null, null, null, false}
+        };
+    }
+
+    @Test(dataProvider = "TestGetIntervalsProvider")
+    public void testGetIntervals(@Nullable String intervals, Integer expected, String ref, boolean expectUserException) {
+        List<String> args = new ArrayList<>(Arrays.asList(
+                "-V", baseVariants
+        ));
+
+        if (ref != null) {
+            args.add("-R");
+            args.add(ref);
+        }
+
+        if (intervals != null) {
+            args.add("-L");
+            args.add(intervals);
+        }
+
+        try {
+            final TestIntervalWalker tool = new TestIntervalWalker();
+            tool.instanceMain(args.toArray(new String[args.size()]));
+
+            Assert.assertEquals(tool.getTraversalIntervals() == null ? null : tool.getTraversalIntervals().size(), expected);
+        }
+        catch (UserException e) {
```

Comment:
We generally try to not intermix cases that throw with cases that don't in the same method (and we generally use either the `expectedExceptions` property of the `@Test` annotation or occasionally`Assert.assertThrows`, to declare methods that throw). But in this case if you remove the test case I mentioned above, the whole try/catch block and expectUserException arg can be removed.